packages/openapi/api/definitions/studios.yaml

```
@@ -0,0 +1,556 @@
+                    items:
+                      type: string
+        404:
+          $ref: '#/components/responses/studioNotFound'
+        500:
+          $ref: '#/components/responses/internalServerError'
+    put:
+      operationId: attachDevice
+      tags:
+        - studios
+      summary: Attaches a device to a studio.
+      parameters:
+        - name: studioId
+          in: path
+          description: Studio to attach the device to.
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Contains the device Id to attach.
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                deviceId:
+                  type: string
+              required:
+                - deviceId
+      responses:
+        200:
+          description: Device successfully attached to the studio.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: number
+                    example: 200
+        404:
+          description: Specified Studio or device does not exist.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: number
+                    example: 404
+                  message:
+                    type: string
+        412:
+          description: The specified device is already attached to a different studio.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: number
+                    example: 412
+                  message:
+                    type: string
+        500:
+          $ref: '#/components/responses/internalServerError'
+  device:
+    delete:
+      operationId: detachDevice
+      tags:
+        - studios
+      summary: Detaches a device from a studio.
+      parameters:
+        - name: studioId
+          in: path
+          description: Studio the device belongs to.
+          required: true
+          schema:
+            type: string
+        - name: deviceId
+          in: path
+          description: Device to remove.
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: Device detached from studio.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: number
+                    example: 200
+        404:
+          $ref: '#/components/responses/studioNotFound'
+        500:
+          $ref: '#/components/responses/internalServerError'
+  studioAction:
+    put:
```

This also doesn't seem like an idempotent action, so maybe this should be `POST`?
```suggestion
    post:
```